### PR TITLE
Fix for BTS-852.

### PR DESCRIPTION
**Pull request by adityamukho**
### Scope & Purpose

Fixes BTS-852 (user's saved queries used to disappear after updating user profile).

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.10: https://github.com/arangodb/arangodb/pull/16676
  - [x] Backport for 3.9: https://github.com/arangodb/arangodb/pull/16677
  - [x] Backport for 3.8: https://github.com/arangodb/arangodb/pull/16678

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-852
- [ ] Design document:

### CODE REVIEW

**Review by hkernbach — APPROVED** [review]

**Review by ajurdak — APPROVED** [review]
LGTM.

File: arangod/RestHandler/RestUsersHandler.cpp [hunk 1]
```diff
@@ -295,7 +297,8 @@ static Result StoreUser(auth::UserManager* um, int mode,
         }
       }
       if (extra.isObject() && !extra.isEmptyObject()) {
-        entry.setUserData(VPackBuilder(extra));
+        VPackSlice oldExtra = u.get("extra");
```
Comment (hkernbach) [inline arangod/RestHandler/RestUsersHandler.cpp:300]: We might need to check here if "oldExtra" is an object. What happens if it is a string or a numeric value and gets merged then by "VPackCollection::merge"?

**Review by hkernbach — COMMENTED** [review]
Changelog entry is missing. Also there might be the need of checking both, old and new, vpack types before we merge them. Otherwise LGTM.

File: arangod/RestHandler/RestUsersHandler.cpp (hunk 1 above)
Comment (adityamukho) [reply arangod/RestHandler/RestUsersHandler.cpp]: Fixed in https://github.com/arangodb/arangodb/pull/16675/commits/22e2f0e6b470bc0b0d5d14ddd0e5b10a661b5b99

File: CHANGELOG [hunk 2]
```diff
@@ -43,7 +43,7 @@ devel
```
[the remark this answers is not in the document]
Comment (adityamukho) [reply CHANGELOG:49]: Fixed in https://github.com/arangodb/arangodb/pull/16675/commits/43c6547ee923378c5ad9191714577112b8aba335

**Comment by adityamukho** [comment]
Jenkins failed at (hopefully unrelated) replication-2 and rebalance tests.

https://jenkins01.arangodb.biz/job/arangodb-matrix-pr-linux.x86-64/24922/EDITION=community,STORAGE_ENGINE=rocksdb,TEST_SUITE=cluster,limit=linux&&test&&x86-64/artifact/testfailures.txt/*view*/

https://jenkins01.arangodb.biz/job/arangodb-matrix-pr-linux.aarch64/3470/EDITION=enterprise,STORAGE_ENGINE=rocksdb,TEST_SUITE=cluster,limit=linux&&test&&aarch64/artifact/testfailures.txt/*view*/

**Review by hkernbach — APPROVED** [review]
LGTM :)

**Pull request merged**